src/util/blockchain.ts

```diff
@@ -460,6 +467,48 @@ export const blockchainForChainIdV2 = (chainId?: number, network = Network.MainN
   }
 }
 
-export const EvmChains = ['Ethereum', 'Binance Smart Chain', 'Arbitrum', 'Polygon', 'OKC', 'Carbon'] as const;
+// function to map hydrogen chain to BlockchainV2 format
+export const blockchainForChainName: { [key: string]: string } = {
+  [Blockchain.Neo]: "Neo",
+  [Blockchain.Ethereum]: "Ethereum",
+  [Blockchain.BinanceSmartChain]: "Binance Smart Chain",
+  [Blockchain.Zilliqa]: "Zilliqa",
+  [Blockchain.Arbitrum]: "Arbitrum",
+  [Blockchain.Polygon]: "Polygon",
+  [Blockchain.Okc]: "OKC",
+  [Blockchain.Native]: "Native",
+  [Blockchain.Btc]: "BTC",
+  [Blockchain.Carbon]: "Carbon",
+  [Blockchain.Switcheo]: "Switcheo",
+  [Blockchain.TradeHub]: "TradeHub",
+  [Blockchain.PolyNetwork]: "PolyNetwork",
+  [Blockchain.Neo3]: "Neo3",
+  [Blockchain.Osmosis]: "Osmosis",
+  [Blockchain.Ibc]: "IBC",
+  [Blockchain.Terra]: "Terra",
+  [Blockchain.CosmosHub]: "Cosmos Hub",
+  [Blockchain.Juno]: "Juno",
+  [Blockchain.Evmos]: "Evmos",
+  [Blockchain.Axelar]: "Axelar",
+  [Blockchain.Stride]: "Stride",
+  [Blockchain.Kujira]: "Kujira",
+  [Blockchain.Terra2]: "Terra 2",
+  [Blockchain.Quicksilver]: "Quicksilver",
+  [Blockchain.Comdex]: "Comdex",
+  [Blockchain.StafiHub]: "Stafi Hub",
+  [Blockchain.Persistence]: "Persistence",
+  [Blockchain.Stargaze]: "Stargaze",
+  [Blockchain.Canto]: "Canto",
+  [Blockchain.OmniFlixHub]: "OmniFlix Hub",
+  [Blockchain.Agoric]: "Agoric",
+  [Blockchain.Sommelier]: "Sommelier",
+  [Blockchain.Mantle]: "Mantle",
+}
+
+export const getFormattedBlockchainName = (chain: string): BlockchainV2 => {
```

Comment:
Think this should return `undefined` in the case the corresponding `BlockchainV2` chain name is not found:
```typescript
export const getFormattedBlockchainName = (chain: string): BlockchainV2 | undefined => {
  return blockchainForChainName[chain]
}
```